File: 🟢 Easy/2022 - Convert 1D Array Into 2D Array/_2022Convert1DArrayInto2DArray.cpp
class Solution {
public:
    vector<vector<int>> construct2DArray(vector<int>& original, int m, int n) {
        if (m * n != original.size()) {
            return {};
        }

        vector<vector<int>> twoDArr(m, vector<int>(n));
        int k = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                twoDArr[i][j] = original[k++];
            }
        }

        return twoDArr;
    }
};

File: 🟢 Easy/2022 - Convert 1D Array Into 2D Array/_2022Convert1DArrayInto2DArray.js
/**
 * @param {number[]} original
 * @param {number} m
 * @param {number} n
 * @return {number[][]}
 */

var construct2DArray = function (original, m, n) {
    if (m * n != original.length) {
        return [];
    }

    let twoDArr = new Array(m).fill(0).map(() => new Array(n).fill(0));
    let k = 0;
    for (let i = 0; i < m; i++) {
        for (let j = 0; j < n; j++) {
            twoDArr[i][j] = original[k++];
        }
    }

    return twoDArr;
};